iib/web/models.py

```diff
@@ -1398,3 +1407,28 @@ def validate_request_params(request_params, required_params, optional_params):
             and not request_params[param]
         ):
             del request_params[param]
+
+
+def validate_registry_auths(registry_auths):
+    """
+    Validate registry_auths for a build request.
+
+    Only auth item in dockerconfig.json is supported for iib.
+
+    :param dict registry_auths: User provided dockerconfig for authentication
+      to private registries
+    :raises ValidationError: if registry_auths are not in valid format
+    """
+    auths = 'auths'
+    if not isinstance(registry_auths, dict):
+        raise ValidationError('"registry_auths" must be a dict')
+    if list(registry_auths.keys()) != [auths]:
+        raise ValidationError('"registry_auths" must contain single key "auths"')
+    if not registry_auths[auths] or not isinstance(registry_auths[auths], dict):
+        raise ValidationError('"registry_auths.auths" must be a non-empty dict')
```

Comment:
Let's use the variable `auths` in the error message too.

Reply:
Variable added yo err msg.